scripts/apps/authoring/authoring/directives/ItemCarouselDirective.js

```diff
@@ -7,14 +7,15 @@ import {waitForImagesToLoad, waitForAudioAndVideoToLoadMetadata} from 'core/help
 
 const carouselContainerSelector = '.sd-media-carousel__content';
 
+let previousItemsString = null;
```

Comment:
You can't put it here, because it will break in cases where there are multiple instances of the directive in a single page(and also when one article is closed and another opened this will be cached). I think you have to attach it to `scope`.

Reply:
Make sure the scope isn't shared with the controller [like this](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/datetime/reldate-directive.js#L12) or it will break when you have multiple instanced of that directive on the same controller.

Reply:
@tomaskikutis you mean that I should put `scope: { items: '=' }` because that's already done.

Reply:
scope: { itemsSinceLastCarouselRerender: '=' }

Reply:
or `scope: { previousItemsString: '=' }` and then `scope.previousItemsString = itemsString;`

Reply:
I don't think it has to be on scope, link is called for each directive instance so local variable should be fine

Reply:
@petrjasek If you have two instances of the directive and you write the "local" variable, it will update it in all directives, isn't it so?

Reply:
I mean if you wrote to the variable local to the `link` function than it would be really local, but if you write it and the top of the file, it's scoped to the module and isn't really local

Reply:
yep local I mean inside link function, otherwise it's shared

Reply:
If `link` is called only once per directive then I agree with your original point. I thought it was like react's `render` where you can't store local variables which live longer than a single render.

Reply:
right it's called only once, not like render